account_invoice_export/models/account_move.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+import requests
+
+import odoo
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+from odoo.addons.queue_job.job import job
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    invoice_sent_through_http = fields.Boolean(copy=False)
+
+    def send_through_http(self):
+        for invoice in self:
+            if invoice.transmit_method_id.send_through_http:
+                invoice.with_delay()._send_through_http()
+
+    @job
+    def _send_through_http(self):
+        """Send invoice to server configured in transmit method."""
+        self.ensure_one()
+        if self.invoice_sent_through_http:
+            return "Invoice has already been sent through http before."
+        if not self.transmit_method_id.send_through_http:
+            raise UserError(_("Transmit method is not configured to send through HTTP"))
+        file_data = self._get_file_for_transmission_method()
+        headers = self.transmit_method_id.get_transmission_http_header()
+        res = requests.post(
+            self.transmit_method_id.destination_url, headers=headers, files=file_data
+        )
+        if res.status_code != 200:
+            values = {
+                "job_id": self.env.context.get("job_uuid"),
+                "send_error": res.status_code,
+                "transmit_method_name": self.transmit_method_id.name,
+            }
+            with odoo.api.Environment.manage():
+                with odoo.registry(self.env.cr.dbname).cursor() as new_cr:
+                    # Create a new environment with new cursor database
+                    new_env = odoo.api.Environment(
+                        new_cr, self.env.uid, self.env.context
+                    )
+                    # The chatter of the invoice need to be updated, when the job fails
+                    self.with_env(new_env).log_error_sending_invoice(values)
+            raise UserError(
+                _(
+                    "HTTP error {} sending invoice to {}".format(
+                        res.status_code, self.transmit_method_id.name
+                    )
+                )
+            )
+        self.invoice_sent_through_http = True
+        self.invoice_send = True
+        self.log_success_sending_invoice()
```

Comment:
why do not wrap it in a method like handle_invoive_state()

Reply:
I think it is easier and clearer to have one for logging error and another one for success